packages/core/src/domain/session/oldCookiesMigration.ts

```diff
@@ -31,6 +34,12 @@ export function tryOldCookiesMigration(options: CookieOptions) {
     if (oldRumType && /^[012]$/.test(oldRumType)) {
       session[RUM_SESSION_KEY] = oldRumType
     }
-    persistSessionCookie(session, options)
+
+    if (!isSessionInExpiredState(session)) {
+      session.expire = String(dateNow() + SESSION_EXPIRATION_DELAY)
+      persistSessionCookie(options)(session)
+    } else {
+      deleteSessionCookie(options)()
+    }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏reverting this change does not seem to break any test, what is going on there?